compiler/codegen/Relocation.cpp

```diff
@@ -423,6 +423,43 @@ const char *TR::ExternalRelocation::_externalRelocationTargetKindNames[TR_NumExt
    "TR_DebugCounter (59)",
    "TR_ClassUnloadAssumption (60)",
    "TR_J2IVirtualThunkPointer (61)",
+   "TR_InlinedAbstractMethodWithNopGuard (62)",
+   "TR_ValidateRootClass (63)",
+   "TR_ValidateClassByName (64)",
+   "TR_ValidateProfiledClass (65)",
+   "TR_ValidateClassFromCP (66)",
+   "TR_ValidateDefiningClassFromCP (67)",
+   "TR_ValidateStaticClassFromCP (68)",
+   "TR_ValidateClassFromMethod (69)",
+   "TR_ValidateComponentClassFromArrayClass (70)",
+   "TR_ValidateArrayClassFromComponentClass (71)",
+   "TR_ValidateSuperClassFromClass (72)",
+   "TR_ValidateClassInstanceOfClass (73)",
+   "TR_ValidateSystemClassByName (74)",
+   "TR_ValidateClassFromITableIndexCP (75)",
+   "TR_ValidateDeclaringClassFromFieldOrStatic (76)",
+   "TR_ValidateClassClass (77)",
+   "TR_ValidateConcreteSubClassFromClass (78)",
+   "TR_ValidateClassChain (79)",
+   "TR_ValidateRomClass (80)",
+   "TR_ValidatePrimitiveClass (81)",
+   "TR_ValidateMethodFromInlinedSite (82)",
+   "TR_ValidateMethodByName (83)",
+   "TR_ValidateMethodFromClass (84)",
+   "TR_ValidateStaticMethodFromCP (85)",
+   "TR_ValidateSpecialMethodFromCP (86)",
+   "TR_ValidateVirtualMethodFromCP (87)",
+   "TR_ValidateVirtualMethodFromOffset (88)",
+   "TR_ValidateInterfaceMethodFromCP (89)",
+   "TR_ValidateMethodFromClassAndSig (90)",
+   "TR_ValidateStackWalkerMaySkipFramesRecord (91)",
+   "TR_ValidateArrayClassFromJavaVM (92)",
+   "TR_ValidateClassInfoIsInitialized (93)",
+   "TR_ValidateMethodFromSingleImplementer (94)",
+   "TR_ValidateMethodFromSingleInterfaceImplementer (95)",
+   "TR_ValidateMethodFromSingleAbstractImplementer (96)",
+   "TR_ValidateImproperInterfaceMethodFromCP (97)",
+   "TR_SymbolFromManager (98)",
```

Comment:
impressive addition to the list...

Reply:
Is TR_SymbolFromManager actually a validation relocation type?

Reply:
i confess i'm pretty worried that this is going to result in pretty extreme overkill in the common cases we're hopefully already getting right without this degree of complexity

Reply:
> Is TR_SymbolFromManager actually a validation relocation type?

No it's a relocation record, it will patch a pointer and add class unload/redefinition assumptions.


>i confess i'm pretty worried that this is going to result in pretty extreme overkill in the common cases we're hopefully already getting right without this degree of complexity

I understand the concerns; when I first started this work (namely try to improve AOT performance), I wanted to not have to go down this road because of the obvious risks of adding this much code. However, it quickly became apparent to me that for most cases I couldn't, in any reasonable amount of time, convince myself that a particular code path was safe to enable under AOT. Additionally, as the code stands, it becomes very difficult to deal with any code path that works with methods or classes that have a cpIndex of -1. Overall, in order to make AOT better in the long run, I decided to go down this road to make life easier for myself and anyone else in the future dealing with AOT. That's why @jdmpapin and myself came up with this approach, which in essence is a symbol table used by linkers for relocations, to simply the compiler; our goal was to ensure that no one ever has to say "we didn't enable this code under AOT because it is too hard".